djangocms_navigation/migrations/0007_language_migration.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import logging
+
+from django.db import migrations
+
+from cms.utils.i18n import get_current_language
+
+logger = logging.getLogger(__name__)
+
+
+def add_navigation_languages(apps, schema_editor):
+    db_alias = schema_editor.connection.alias
+    navigation_queryset = apps.get_model("djangocms_navigation", "MenuContent").using(db_alias).filter(language_exact="")
+    for model in navigation_queryset:
+        model.language = get_current_language()

```

Comment:
This needs to be done on a per site bases, there is a method that helps to get the language for the site, I saw it yesterday, I'll comment below when I find it again.